File: starter_kits/Rust/src/hlt/shipyard.rs
use hlt::command::Command;
use hlt::entity::Entity;
use hlt::PlayerId;
use hlt::position::Position;

pub struct Shipyard {
    pub owner: PlayerId,
    pub position: Position,
}

impl Shipyard {
    pub fn spawn(&self) -> Command {
        Command::spawn_ship()
    }
}

impl Entity for Shipyard {
    fn owner(&self) -> PlayerId {
        self.owner
    }

    fn position(&self) -> Position {
        self.position
    }
}

File: starter_kits/PHP/hlt/Shipyard.php
<?php

namespace hlt;

class Shipyard extends Entity {
    public function __construct( PlayerId $owner,  Position $position)
    {
        parent::__construct($owner, new EntityId(), $position);
    }

    /**
     * @return Command
     */
    public function spawn() {
        return Command::spawnShip();
    }
}
